src/main/java/thewaypointers/trafficsimulator/gui/PanelMouseAction.java

```diff
@@ -44,11 +43,11 @@ public void mouseClicked(MouseEvent e) {
 
             if (((junction_x <= x) && (x <= junction_x + 50)) && ((junction_y <= y) && (y <= junction_y + 50))) {
                 //pass label here
-                System.out.println("click test, pass label to the simulation");
-
+                //System.out.println("click test, pass label to the simulation");
+                SimulationInputListener s = new SimulationInputListener();
```

Comment:
This still doesn't really solve the problem, because the input listeners (for example, the actual simulation) are created somewhere else and we would like to use them here, and you can't easily do that. I think the `SimulationInputListener` should be passed from outside, to be the receiver of all events happening in the control panel or GUI.